### ISSUE TITLE
List ^ ZZ for permutations

### ISSUE THREAD

**Issue by mahrud** [Macaulay2/M2#2581]
Here's something that might be nice for combinatorialists working with permutations:
```m2
i1 : w = {2,0,1}

o1 = {2, 0, 1}

o1 : List

i2 : w^2 -- essentially w_w

o2 = {1, 2, 0}

o2 : List

i3 : w^3 -- essentially (w_w)_w

o3 = {0, 1, 2}

o3 : List

i4 : w^-1 -- essentially inversePermutation w

o4 = {1, 2, 0}

o4 : List

i5 : w^-2 -- essentially (inversePermutation w)_(inversePermutation w)

o5 = {2, 0, 1}

o5 : List
```
Here's a simple implementation:
```m2
List ^ ZZ := List => (w, n) -> fold(
    if n < 0 then (-n):inversePermutation w else n:w,
    toList(0 ..< #w),
    (w, v) -> v_w)
```
What's nice is that if the entries are not integers `List _ List` handles the errors correctly.

**Comment by DanGrayson** [comment]
Good idea, but introduce a new type of list called "Permutation" to implement these things for.

**Comment by mahrud** [comment]
> Good idea, but introduce a new type of list called "Permutation" to implement these things for.

Part of the goal is to appeal to people closer to combinatorics by keeping the syntax as close to the standard syntax as possible (in the same way that, for instance, `QQ[x,y,z]` appeals to others).

**Comment by TheGrateSalmon** [comment]
I have written some code to resolve this and add some other permutation methods (e.g., inversions), so there should be a pull request in the next few weeks (hopefully). 

However, I think there is a convention choice that has to be made. Namely, in any math that I have seen, permutations are 1-indexed, but functionally in code, permutations are 0-indexed; i.e., should permutations be permutations on 0..n-1 or 1..n? While helper functions to go between the two conventions is possible so that the user has choice in which convention they use, this seems very unwieldy. Are there any thoughts on this?

**Comment by mahrud** [comment]
I think we should go with the 0-indexed convention, because there's actually a very easy way to convert the 1-indexed notation into 0-based: simply prepend a 0 and in the end drop the 0!

Here's what I mean: let's say you want to check that (3,1,2)^2 is (2,3,1) in the 1-based notation, you can do it this way:
```m2
i3 : w = {0,3,1,2}; w_w

o4 = {0, 2, 3, 1}
```
